app/components/OutcomeBar.tsx

```diff
@@ -1,64 +1,136 @@
-'use client';
-
-import { FixedProductMarketMaker } from '@/queries/omen';
 import { cx } from 'class-variance-authority';
+import { useQuery } from '@tanstack/react-query';
+import request from 'graphql-request';
+
+import { BlockDataType } from '@/queries/xdai';
+import {
+  FixedProductMarketMaker,
+  FpmmTrade_OrderBy,
+  getMarketTrades,
+} from '@/queries/omen';
+import { OMEN_SUBGRAPH_URL, XDAI_BLOCKS_SUBGRAPH_URL } from '@/constants';
 import { Outcome } from '@/entities';
 
 interface OutcomeBarProps {
   market: FixedProductMarketMaker;
 }
 
+type OutcomeTokenMarginalPricesResponse = {
+  [key: string]: {
+    outcomeTokenMarginalPrices: string[];
+  };
+};
+
 export const OutcomeBar = ({ market }: OutcomeBarProps) => {
+  const { id } = market;
+
+  const { data: trades } = useQuery({
+    queryKey: ['getMarketTrades', id],
+    queryFn: async () =>
+      getMarketTrades({
+        fpmm: id,
+        orderBy: FpmmTrade_OrderBy.CreationTimestamp,
+      }),
+  });
+
+  const { data: lastTradeMarginalPrices } = useQuery({
+    queryKey: ['getLastTradeMarginalPrices', id],
+    queryFn: async (): Promise<string[] | undefined> => {
+      if (!trades) return;
+
+      const lastTradeTimestamp =
+        trades.fpmmTrades[trades.fpmmTrades.length - 1]?.creationTimestamp;
+
+      const blockNumber = await request<BlockDataType>(
+        XDAI_BLOCKS_SUBGRAPH_URL,
+        `query blockNumberByTimestamp {
+          _${lastTradeTimestamp}: blocks(where: {timestamp: ${lastTradeTimestamp}}, first: 1) {
+              number
+            }
+        }`
+      );
+
+      const tradeBlockNumber = blockNumber[`_${lastTradeTimestamp}`][0].number;
+
+      if (!tradeBlockNumber) return;
+
+      const marginalPricesResponse = await request<OutcomeTokenMarginalPricesResponse>(
+        OMEN_SUBGRAPH_URL,
+        `query marginalPricesByBlockNumber($id: ID!) {
+         _${tradeBlockNumber}: fixedProductMarketMaker(id: $id, block: { number: ${tradeBlockNumber} }) {
+              outcomeTokenMarginalPrices
+            }
+        }`,
+        {
+          id,
+        }
+      );
```

Comment:
Similar to the fix above. `OutcomeTokenMarginalPricesResponse` will be different.

```suggestion
      const marginalPricesResponse = await request<OutcomeTokenMarginalPricesResponse>(
        OMEN_SUBGRAPH_URL,
        `query marginalPricesByBlockNumber($id: ID!, $tradeBlockNumber: Int) {
            fixedProductMarketMaker(id: $id, block: { number: $tradeBlockNumber }) {
              outcomeTokenMarginalPrices
            }
        }`,
        {
          id,
          tradeBlockNumber
        }
      );
```

Reply:
Noted! Doing it